src/client/api/pythonApi.ts

```diff
@@ -107,17 +115,19 @@ export class PythonExtensionChecker implements IPythonExtensionChecker {
         return this.extensions.getExtension(this.pythonExtensionId) !== undefined;
     }
 
-    public async showPythonExtensionInstallRequiredPrompt(): Promise<void> {
+    public async showPythonExtensionInstallRequiredPrompt(): Promise<InstallerResponse> {
         if (this.waitingOnInstallPrompt) {
-            return this.waitingOnInstallPrompt;
+            return InstallerResponse.Waiting;
```

Comment:
Why can you just return `this.waitingOnInstallPrompt`

& have `this.waitingOnInstallPrompt` get resolved with the value of Installerresponse.

Reply:
Basically returning `Waiting` is unnecessary, we can just wait on this promise, then its not `Waiting` anymore and we'd have a result.